hoomd/BoxResizeUpdater.cc

```diff
@@ -45,105 +39,134 @@ BoxResizeUpdater::~BoxResizeUpdater()
     m_exec_conf->msg->notice(5) << "Destroying BoxResizeUpdater" << endl;
     }
 
-/*! \param scale_particles Set to true to scale particles with the box. Set to false to leave particle positions alone
-    when scaling the box.
-*/
-void BoxResizeUpdater::setParams(bool scale_particles)
+/// Get the current box based on the timestep
+BoxDim BoxResizeUpdater::getCurrentBox(unsigned int timestep)
     {
-    m_scale_particles = scale_particles;
+    Scalar min = m_variant->min();
+    Scalar max = m_variant->max();
+    Scalar cur_value = (*m_variant)(timestep);
+    Scalar scale = 0;
+    if (cur_value == max)
+        {
+        scale = 1;
+        }
+    else if (cur_value > min)
+        {
+        scale = (cur_value - min) / (max - min);
+        }
+
+    auto initial_box = getBoxDimFromPyObject(m_initial_box);
+    auto final_box = getBoxDimFromPyObject(m_final_box);
```

Comment:
The problem come in when the user does something like `box_resize.box1.L = 10`. This should change the internal box. Currently we export the `BoxDim` class wrapped in a `std::unique_ptr`. This prevents having multiple full pointers, so we have to copy to store the `BoxDim` object directly in C++. We could change this to a `std::shared_ptr` like we do elsewhere, I just don't know if this would break existing code.

Reply:
I see, so right now the Python box object is the single source of truth. I'm not sure where you're referring to the `unique_ptr`, but since the `getBoxDimFromPyObject` is just pulling the pointer to the `_cpp_obj` from the Python box, what exactly is the type of `_cpp_obj`? I don't know the pybind API very well, is it a pointer to the `BoxDim` object, a copy of the object, or is it wrapped in some intermediate layer? Is it feasible to just return a `const BoxDim [*|&]` from `getBoxDimFromPyObject` so that you don't make new objects each time this is called?